iot-hub/Samples/device/MessageReceiveSample/MessageReceiveSample.cs

```diff
@@ -71,6 +70,7 @@ private async Task ReceiveC2dMessagesPollingAndComplete(TimeSpan timeout)
 
                 await _deviceClient.CompleteAsync(receivedMessage);
                 Console.WriteLine($"{DateTime.Now}> Completed C2D message with Id={receivedMessage.MessageId}.");
+                break;
```

Comment:
this break isn't required, right? we want to keep polling until we time out, and not just on 1st message received.

Reply:
I found the sample flow confusing otherwise. I sent a message that was picked up via polling, and was hoping to move on to the next one. Instead I just get to wait.

Well, if you are waiting 10 seconds for the whole operation then it isn't too long, but as I mentioned in another comment, 10 seconds wasn't enough for me to get it done.

So I figured, giving you 1 minute to get a C2D message off, and then proceed to the next part of the sample was more interesting.

Thoughts?

Reply:
I can get on board with that. 
The idea initially was to show the common prod scenario, where applications would continuously poll for messages and process them as received. This not running on a separate thread but rather waiting for this polling to complete to continue to other scenarios kind-of deviates from that idea anyway.

I'd recommend updating the method name and associated comments to indicate that this method will process a single message:
Device waiting for C2D messages from the hub for {s_receiveTimeout}... -> Device waiting for a C2D message from the hub for {s_receiveTimeout}...
ReceiveC2dMessagesPollingAndComplete -> ReceiveC2dMessagePollingAndComplete

Reply:
What if we changed it to take user input to move to the next phase? That would mean no timeout. It says in Phase 1 until told to move to Phase 2, and then same in Phase 2 until time to quit?

Reply:
Yeah, that'd be good. I think that'll tie in well with the whole concept of the polling receive API.
Small note - the default case will need to have a timeout (customizable), just because we want this sample to run as a part of our nightly pipeline run.

Reply:
Updated. Take a look! :)